extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.ads.js

```diff
@@ -0,0 +1,95 @@
+define('wikia.articleVideo.featuredVideo.ads', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.video.vastUrlBuilder',
+	'ext.wikia.adEngine.slot.service.megaAdUnitBuilder',
+	'wikia.log'
+], function (adContext, vastUrlBuilder, megaAdUnitBuilder, log) {
+
+	var aspectRatio = 640 / 480,
+		logGroup = 'wikia.articleVideo.featuredVideo.ads';
+
+	function calculateRV(depth) {
+		var capping = adContext.get('opts.fvAdsFrequency');
+
+		return depth < 2 || !capping ? 1 : Math.floor((depth - 1) / capping) + 1;
+	}
+
+	function cappingAllowsToShowNextVideo(depth) {
+		var capping = adContext.get('opts.fvAdsFrequency');
+
+		return adContext.get('opts.replayAdsForFV') && capping > 0 && (depth - 1) % capping === 0;
+	}
+
+	function canAdBePlayed(depth) {
+		var isReplay = depth > 1;
+
+		return !isReplay || (isReplay && cappingAllowsToShowNextVideo(depth));
+	}
+
+	function shouldPlayPreroll(videoDepth) {
+		return canAdBePlayed(videoDepth);
+	}
+
+	function shouldPlayMidroll(videoDepth) {
+		return adContext.get('opts.isFVMidrollEnabled') && canAdBePlayed(videoDepth);
+	}
+
+	function shouldPlayPostroll(videoDepth) {
+		return adContext.get('opts.isFVPostrollEnabled') && canAdBePlayed(videoDepth);
+	}
+
+	function buildVastUrl(position, videoDepth, correlator) {
+		var options = {
+				vpos: position,
+				correlator: correlator
+			},
+			slotParams = {
+				passback: 'jwplayer',
+				pos: 'FEATURED',
+				rv: calculateRV(videoDepth),
+				src: 'premium'
+			};
+
+		if (adContext.get('opts.megaAdUnitBuilderEnabled')) {
```

Comment:
I believe we have MEGA enabled for all FV pages. Do we still need this check?

Reply:
megaAdUnitBuilderEnabled flag depends on premium ad layout as well

Reply:
Isn't PAL enabled on all FV pages?

Reply:
You're right. Asked Nef